webdriver-extension/index.bs

```diff
@@ -0,0 +1,535 @@
+    text: url variable; url: dfn-url-variables
+    text: session; url: dfn-session
+    text: success; url: dfn-success
+    text: handling errors
+    text: Object; url: dfn-object
+urlPrefix: https://w3c.github.io/sensors/; spec: GENERIC-SENSOR
+  type: dfn
+    text: platform sensor; url: concept-platform-sensor
+    text: sensor
+    text: sensor reading
+    text: sensor type
+    text: reading timestamp
+    text: latest reading
+    text: connect to sensor
+    text: sampling frequency
+    text: requested sampling frequency
+    text: can expose sensor readings
+    text: update latest reading
+urlPrefix: https://w3c.github.io/ambient-light; spec: AMBIENT-LIGHT
+  type: dfn
+    text: AmbientLightSensor; url: ambient-light-sensor-interface
+urlPrefix: https://w3c.github.io/accelerometer; spec: ACCELEROMETER
+  type: dfn
+    text: Accelerometer; url: accelerometer-interface
+    text: LinearAccelerationSensor; url: linearaccelerationsensor-interface
+    text: GravitySensor; url: gravitysensor-interface
+urlPrefix: https://w3c.github.io/gyroscope; spec: GYROSCOPE
+  type: dfn
+    text: Gyroscope; url: gyroscope-interface
+urlPrefix: https://w3c.github.io/magnetometer; spec: MAGNETOMETER
+  type: dfn
+    text: Magnetometer; url: magnetometer-interface
+    text: UncalibratedMagnetometer; url: uncalibrated-magnetometer-interface
+urlPrefix: https://w3c.github.io/orientation-sensor; spec: ORIENTATION-SENSOR
+  type: dfn
+    text: AbsoluteOrientationSensor; url: absoluteorientationsensor-interface
+    text: RelativeOrientationSensor; url: relativeorientationsensor-interface
+urlPrefix: https://wicg.github.io/geolocation-sensor/; spec: GEOLOCATION-SENSOR
+  type: dfn
+    text: GeolocationSensor; url: geolocationsensor-interface
+urlPrefix: https://w3c.github.io/proximity; spec: PROXIMITY
+  type: dfn
+    text: ProximitySensor; url: proximity-sensor-interface
+</pre>
+
+Introduction {#intro}
+=====================
+
+Generic Sensor API [[GENERIC-SENSOR]] and its concrete Sensor APIs pose a challenge
+to test authors because to fully exercise their interfaces requires physical hardware
+devices that respond in predictable ways. To address this challenge this specification
+defines <a>extension commands</a> to the [[WebDriver]] specification for controlling
+[=mock sensor=] on the host that the user agent is running on.
+With these <a>extension commands</a> devices with particular properties can be created
+and their responses to requests are well defined.
+
+Mock Sensors {#mock-sensors}
+=====================
+
+A <dfn id="mock-sensor">mock sensor</dfn> simulates the behavior of a <a>platform sensor</a> in controlled ways.
+
+A [=mock sensor=] reports corresponding <dfn id="mock-sensor-reading">mock <a>sensor reading</a></dfn>, which is a source of
+mocking information about the environment, to the <a>Sensor</a> objects.
+
+Note: It's up to the user agent to expose what kinds of <a>mock sensor reading</a> to the <a>Sensor</a> objects.
```

Comment:
I think I understand. It sounds like the UA doesn't get to choose the *kind* of
reading, but rather the reading itself (or more precisely, its value). Is that
right? If so, we should remove the phrase "what kinds of":

> Note: It's up to the user agent to expose mock sensor reading[s] to the
> Sensor objects.

Is that accurate? It's a little vague of me, though--of course the UA has to
expose readings! It might be more precise/helpful to say something like:

> Note: The user agent must select the mock sensor readings that are initially
> exposed to the Sensor objects.

Reply:
Yep, you are right. How about use "provide"(or maybe "set") instead of "select"?

> Note: The user agent must provide the mock sensor readings that are initially
exposed to the Sensor objects.

Reply:
Works for me!